dot/parachain/collator-protocol/validator_side_test.go

```diff
@@ -411,30 +426,30 @@ func TestProcessBackedOverseerMessage(t *testing.T) {
 		t.Run(c.description, func(t *testing.T) {
 			t.Parallel()
 
-			subsystemToOverseer := make(chan any)
-
 			ctrl := gomock.NewController(t)
 			defer ctrl.Finish()
+			overseer := overseer.NewMockableOverseer(t)
 
 			collationProtocolID := "/6761727661676500000000000000000000000000000000000000000000000000/1/collations/1"
 
 			net := NewMockNetwork(ctrl)
-			net.EXPECT().RegisterNotificationsProtocol(
-				gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(),
-				gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
 			net.EXPECT().GetRequestResponseProtocol(gomock.Any(), collationFetchingRequestTimeout,
 				uint64(collationFetchingMaxResponseSize)).Return(&network.RequestResponseProtocol{})
-			net.EXPECT().GetNetworkEventsChannel().Return(make(chan *network.NetworkEventInfo))
-			cpvs, err := Register(net, protocol.ID(collationProtocolID), subsystemToOverseer)
-			require.NoError(t, err)
+			cpvs := New(net, protocol.ID(collationProtocolID), overseer.GetSubsystemToOverseerChannel())
 
 			cpvs.BlockedAdvertisements = c.blockedAdvertisements
 
-			go func() {
-				msg, _ := (<-subsystemToOverseer).(backing.CanSecondMessage)
-				msg.ResponseCh <- c.canSecond
-			}()
+			cpvs.OverseerToSubSystem = overseer.RegisterSubsystem(cpvs)
+
+			err := overseer.Start()
+			require.NoError(t, err)
+
+			defer overseer.Stop()
+
+			overseer.ReceiveMessage(c.msg)
+			overseer.ExpectActions(c.expectedActions...)
 
+			time.Sleep(1 * time.Second)
```

Comment:
We need overseer here. When we processMessage we will send out `backing.CanSecondMessage`, we need overseer to receive that.

Reply:
We have this code to handle that message. Since you are not running the subsystem as a goroutine, you don't really need an overseer.

```
go func() {
	msg, _ := (<-subsystemToOverseer).(backing.CanSecondMessage)
	msg.ResponseCh <- c.canSecond
}()
```